api/placer.py

```diff
@@ -0,0 +1,291 @@
+import bson
+import copy
+import datetime
+import dateutil
+import os
+import pymongo
+import zipfile
+
+from . import base
+from . import config
+from . import files
+from . import rules
+from . import tempdir as tempfile
+from . import util
+from . import validators
+from .dao import reaperutil, APIStorageException
+
+
+class Placer(object):
+    """
+    Interface for a placer, which knows how to process files and place them where they belong - on disk and database.
+    """
+
+    def __init__(self, container_type, container, id, metadata, timestamp):
+        self.container_type = container_type
```

Comment:
Something we need to consider in the Placer structure is also that there could be multiple target container for a single upload.
@kofalt I will try to address this in the LabelPlacer development, but let me know what you think about this.